static/app/views/dashboards/globalFilter/filterSelector.tsx

```diff
@@ -83,16 +83,13 @@ function FilterSelector({
     ? !fullTag.predefined && predefinedValues === null
     : true;
 
-  const baseQueryKey = useMemo(
-    () => ['global-dashboard-filters-tag-values', tag, selection, searchQuery],
-    [tag, selection, searchQuery]
-  );
-  const queryKey = useDebouncedValue(baseQueryKey);
-
-  const queryResult = useQuery<string[]>({
-    // Disable exhaustive deps because we want to debounce the query key above
-    // eslint-disable-next-line @tanstack/query/exhaustive-deps
-    queryKey,
+  const queryResult = useQuery({
+    queryKey: useDebouncedValue(
+      useMemo(
+        () => ['global-dashboard-filters-tag-values', tag, selection, searchQuery],
+        [tag, selection, searchQuery]
+      )
+    ),
```

Comment:
this is a limitation of the eslint plugin - it doesn’t track the usage through multiple hooks. The easiest way is to just inline the hook calls, which is fine - it doesn’t violate the rules of hooks as they are all still called top-level.

we can do this if we don’t need the debounced intermediate value anywhere else.